config/render.go

```diff
@@ -0,0 +1,75 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+// RenderApp encapsulates the rendered configuration file and the input and
+// output paths.
+type RenderApp struct {
+	renderFlag     string
+	renderedConfig []byte
+}
+
+// EmptyApp creates an empty application
+func EmptyApp() *RenderApp {
+	app := &RenderApp{}
+	return app
+}
+
+// NewRenderApp creates a new Config Rendering application
+func NewRenderApp(configFlag string, renderFlag string) (*RenderApp, error) {
```

Comment:
> I went through the ParseConfig file again and I did not see the "template filter functions" you talked about. Can you be more specific where they are located?

I was talking about [these](https://github.com/joyent/containerpilot/blob/master/config/template.go#L84) but on second look you actually are passing thru those. So the approach you have after https://github.com/joyent/containerpilot/pull/271/commits/06e0dc63843f9e20ca99e19b6f16aae8880c7d52 is fine but let's factor out the [repeated code into functions](https://github.com/joyent/containerpilot/pull/271/files#diff-639ed475181e0558956e92333cd5e9c0R149)